File: PlaywrightTypeScript/tests/page-object-model/shared-components/navbarcomponent.ts
import { Page, Locator } from "@playwright/test";

export class NavbarComponent {
    readonly welcomeLinkLocator: Locator;
    readonly formLinkLocator: Locator;
    readonly exampleResultsLinkLocator: Locator;

    constructor(readonly page: Page) {
        let navbarLocator = page.locator("app-nav");

        this.welcomeLinkLocator = navbarLocator.locator("'Welcome'");
        this.formLinkLocator = navbarLocator.locator("'The Form'");
        this.exampleResultsLinkLocator = navbarLocator.locator("'Example results'");
    }
    
    async gotoWelcome(): Promise<void> {
        await this.welcomeLinkLocator.click();
    }
    
    async gotoForm(): Promise<void> {
        await this.formLinkLocator.click();
    }
    
    async gotoExampleResults(): Promise<void> {
        await this.exampleResultsLinkLocator.click();
    }
}

File: PlaywrightDotNet/PlaywrightDotnet.Tests/PageObjectModel/SharedComponents/NavBarComponent.cs
namespace PlaywrightDotnet.Tests.PageObjectModel.SharedComponents;

public interface INavBarComponent
{
    Task GotoFormAsync();
    Task GotoWelcomeAsync();
}

internal class NavBarComponent : BaseComponent, INavBarComponent
{
    private ILocator NavbarLocator => Page.Locator("app-nav");

    public ILocator ExampleResultsLinkLocator => NavbarLocator.Locator("'Example results'");
    public ILocator FormLinkLocator => NavbarLocator.Locator("'The Form'");
    public ILocator WelcomeLinkLocator => NavbarLocator.Locator("'Welcome'");

    public NavBarComponent(IPage page) : base(page) { }

    public async Task GotoFormAsync()
    {
        await FormLinkLocator.ClickAsync();
    }

    public async Task GotoWelcomeAsync()
    {
        await WelcomeLinkLocator.ClickAsync();
    }
}